### PR TITLE
Partial revert of compute-task message format

### PR DESCRIPTION
**Pull request by fjetter**
This is a hotfix for https://github.com/dask/distributed/issues/6624 by reverting the compute-task message format almost to the original state before https://github.com/dask/distributed/pull/6410

I didn't debug very deeply but it appears that our (de-)serializer cannot handle nested Serialized objects very well or something else is messed up. By formatting the `run_spec` value as a dict the way #6410 did appears to be a compatible change but the nesting is destroyed as soon as it reaches the worker.

basically

```python
# Scheduler side

dct_scheduler["run_spec"] = {"function" b"foo", "args": b"bar", "kwargs": "baz"}

# becomes on worker side

dct_worker["run_spec"] = deserialize(dct_scheduler["args"])
```

### CODE REVIEW

**Comment by fjetter** [comment]
@jakirkham maybe you have an idea what's going on here?

File: distributed/tests/test_scheduler.py [hunk 1]
```diff
@@ -3641,3 +3641,18 @@ async def test_worker_state_unique_regardless_of_address(s, w):
...
+    da = pytest.importorskip("dask.array")
+    np = pytest.importorskip("numpy")
+    with Client():
+        v = da.random.random((20, 20), chunks=(5, 5))
+
+        overlapped = da.map_overlap(np.sum, v, depth=2, boundary="reflect")
+        # This computation is somehow broken but we want to avoid catching any
+        # serialization errors that result in KilledWorker
+        with pytest.raises(IndexError):
+            overlapped.compute()
```
Comment (fjetter) [inline distributed/tests/test_scheduler.py:3658]: This is obviously not ideal but I simply took the reproducer from https://github.com/dask/distributed/issues/6624 
I do not hit the serialization error anymore but rather an IndexError ¯\_(ツ)_/¯

**Comment by fjetter** [comment]
All green 🎉  👀

**Review by jsignell — APPROVED** [review]
This seems like a nice clean solution to me!
